pkg/scheduler/extender/extender.go

```diff
@@ -431,6 +445,42 @@ func (e *Extender) createReservation(ctx context.Context, namespace string, name
 		// cannot create reservation
 		return err
 	}
+	// wait for reservation to be confirmed
+	// continue on reading error
+	// return after reservation confirmed
+	// or wait context deadline
+	readCh := make(chan struct{})
+	ticker := time.NewTicker(time.Second)
+	ctx, cancel := context.WithTimeout(ctx, reservationConfirmedDuration)
```

Comment:
Don't context coming from scheduler without timeout? Why did you chose 5 minutes?

Reply:
> Don't context coming from scheduler without timeout? Why did you chose 5 minutes?

This context comes from [http handler](https://github.com/dell/csi-baremetal/blob/master/pkg/scheduler/extender/extender.go#L110) and there is no deadline or timeout on this context.

I chose 5 minutes because context must have deadline.

Reply:
Do you mean the scheduler sends requests to the extender(s) and don't fail them by timeout?

Reply:
I mean the context in this function doesn't have timeout at all.

Reply:
We can set any timeout from our side but it doesn't make a lot of sense if we aren't align with what scheduler does.

Reply:
Client can have own timeouts not aligned with server.

Reply:
What is the goal of these changes then? It does make sense to wait for response only till scheduler is able to handle response from our side.

Reply:
Every http handler should have timeout so it wont stuck forever. I chose 5 minutes as a start point. Do you have any consideration how long it should be?

Reply:
In this case what is the purpose to have 5 minutes timeout for `FilterHandler`?

Reply:
Because the nature of CSI, I assume some calls of `CR` creation can take much more than 5 seconds (with transition from requested to confirmed).

Reply:
When we patch scheduler configuration we override default `httpTimeout` from 5 to 15 seconds. It means that you don't need to add own timeout - just wait for http context to be cancelled.

Reply:
ok